activesupport/lib/active_support/testing/parallelize_when_suitable_executor.rb

```diff
@@ -0,0 +1,37 @@
+module ActiveSupport
+  module Testing
+    class ParallelizeWhenSuitableExecutor
+      def initialize(parallel_executor)
+        @parallel_executor = parallel_executor
+      end
+
+      def start
+        parallelize if worth_parallelizing?
```

Comment:
I moved the parallelization logic here because we need to run it **after** all the test files are loaded. I did it this way to avoid having to change/patch Minitest. This is just a way of loading the parallelization logic in a lazy way, after all the test cases have been loaded. Happy to change the approach if you can think of a better one.

Reply:
I think I prefer `should_parallelize` over `worth_parallelizing`. "worthy" feels like a value judgement to me.